### PR TITLE
Rework tooltip to be more logical

### LINKED ISSUE #13035: Modify rct_window_event_list::tooltip to return arguments
The `rct_window_event_list::tooltip` function currently returns a `string_id*` for its last parameter. It also passes back formatter arguments via the global variable `gCommonFormatArgs`. The `rct_window_event_list::tooltip` should instead return a `rct_string_id` and a `Formatter`. This reduces the usage of `gCommonFormatArgs` and its associated magic.

`std::pair<rct_string_id, Formatter> (*tooltip)(struct rct_window*, rct_widgetindex)`

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Closes #13035. I'm not 100% convinced on the ride graph tooltips this adds them but they dont format correctly.

### CODE REVIEW

File: src/openrct2-ui/windows/Options.cpp [hunk 1]
```diff
@@ -2107,21 +2107,23 @@ static void window_options_advanced_paint(rct_window* w, rct_drawpixelinfo* dpi)
     DrawTextEllipsised(dpi, screenCoords, 277, STR_STRING, ft, w->colours[1]);
 }
 
-static void window_options_advanced_tooltip(rct_window* w, rct_widgetindex widgetIndex, rct_string_id* stringid)
+static OpenRCT2String window_options_advanced_tooltip(rct_window* w, const rct_widgetindex widgetIndex, const rct_string_id fallback)
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Options.cpp:2110]:
```suggestion
static OpenRCT2String window_options_advanced_tooltip(
    rct_window* w, const rct_widgetindex widgetIndex, const rct_string_id fallback)
```

File: src/openrct2-ui/windows/GameBottomToolbar.cpp [hunk 2]
```diff
@@ -189,10 +189,10 @@ static void window_game_bottom_toolbar_mouseup(rct_window* w, rct_widgetindex wi
     }
 }
 
-static void window_game_bottom_toolbar_tooltip(rct_window* w, rct_widgetindex widgetIndex, rct_string_id* stringId)
+static OpenRCT2String window_game_bottom_toolbar_tooltip(rct_window* w, const rct_widgetindex widgetIndex, const rct_string_id fallback)
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/GameBottomToolbar.cpp:192]:
```suggestion
static OpenRCT2String window_game_bottom_toolbar_tooltip(
    rct_window* w, const rct_widgetindex widgetIndex, const rct_string_id fallback)
```

**Review by tupaschoal — APPROVED** [review]
> Closes #13035. I'm not 100% convinced on the ride graph tooltips this adds them but they dont format correctly.

But had they never worked correctly or at all before? If so I'm in favour of merging this as is and opening a bug issue, if not then I guess we should tackle it here.

Changeset looks fine, though I do find it a bit weird that all methods receive a `fallback` and don't do anything with it other than returning when there is an error.

**Comment by duncanspumpkin** [comment]
They had never worked before.

**Comment by duncanspumpkin** [comment]
The fallback follows the style of the cursor event

**Comment by tupaschoal** [comment]
Should we have a changelog entry for the graph tooltip or do we add it when it's fixed?

**Comment by duncanspumpkin** [comment]
Add it when its fixed. I think doing https://github.com/OpenRCT2/OpenRCT2/issues/12993 will fix it.

**Pull request merged**